### PR TITLE
Record info in next test module after loading snapshot

### PR DESCRIPTION
**Pull request by okurz**
After a test module is rolled back, i.e. a snapshot is loaded, the
system under test can behave different, e.g. not being fully responsive
yet or simply presenting a different starting situation. To make the
potential impact more obvious to test reviewers we now record a specific
test result step informing in the next module that a snapshot was loaded
in before.

Related progress issue: https://progress.opensuse.org/issues/103791

### CODE REVIEW

**Review by Martchus — APPROVED** [review]

**Review by kalikiana — APPROVED** [review]
I think this is looking rather nice now

File: autotest.pm [hunk 1]
```diff
@@ -372,6 +372,7 @@ sub runalltests {
         eval { $t->runtest; };
         my $error = $@;    # save $@, it might be overwritten
         $t->save_test_result();
+        my $next_test = $testindex eq $#testorder ? undef : $testorder[$testindex + 1];
```
Comment (perlpunk) [inline autotest.pm:375]: Why `eq` and not `==`?

Comment (perlpunk) [reply autotest.pm:375]: Just `my $next_test = $testorder[$testindex + 1]` would have worked, too. Why the extra check? It's perl, not python ;-)

Comment (okurz) [reply autotest.pm:375]: Ok, sure. @kalikiana and me were just too cautious ;) Created https://github.com/os-autoinst/os-autoinst/pull/1989

**Pull request merged**